aten/src/ATen/native/TensorShape.cpp

```diff
@@ -115,10 +115,14 @@ Tensor expand_as(const Tensor& self, const Tensor& other) {
 Tensor narrow(const Tensor& self, int64_t dim, int64_t start, int64_t length) {
   AT_CHECK(self.dim() > 0, "narrow() cannot be applied to a 0-dim tensor.");
   auto cur_size = self.size(dim);
-  if (start < 0 || start >= cur_size) {
+  if (start < 0) {
     AT_ERROR("start out of range");
   }
+#ifndef USE_TH_SIZE_ZERO_DIM
   if (length <= 0 || start > cur_size - length) {
+#else
+  if (length < 0 || start > cur_size - length) {
+#endif
     AT_ERROR("length out of range");
```

Comment:
I found this a little surprising. If I say `narrow(self, 0, 999999999, 0)`, where `self.size(0) == 10`, this code will report now "length out of range", which... seems like it's missing the point.

Reply:
(I probably should have put this comment on the if statement change above.)

Reply:
I'll change the message to something like "start + length exceeds dimension size".